### ISSUE TITLE
[Collaboration] Nail Links

### ISSUE THREAD

**Issue by alexkim205** [PostHog/posthog#7245]
This issue provides context about what we (@clarkus @paolodamico @alexkim205) decided to tackle first in the [Collaboration Milestone](https://github.com/PostHog/product-internal/pull/204).

Notes from call are in [Google Docs](https://docs.google.com/document/d/1B96cJg6f0M_eBqA-rm67Po4-_jPPwPGnqfHWJsMCLHE/).

## Links

**Question** - How do we enable our users to **create and share links** to the PH app with internal and external people. 

**Answer** - Permalinks + Centralized link management page + Integrations. 

Permalinks - Non-expiring links to PH pages for internal and external users.
Centralized link management - One place to manage all links that have been created by your team. Can configure url, expiration(?), permissions(?), etc.
Integrations - Links shared in Slack, Discord, Notion, Teams, Google Docs, etc. will be enriched with metadata and preview images.

**Other solutions considered** - No way of creating and sharing links to external users.

## Breakdown

- Considerations
	- What should/shouldn't be available for public insights and dashboards? (i.e., person modal, date pickers, annotations, etc.)
	- Depending on which channel the link is shared, what kind of metadata should we enrich the link with?
- Out-of-scope (possible future improvements)
	- Permissions at the insight level. We currently have permissions at the project level and adding insight-level rules will affect other parts of the app.
	- Link usage metrics.
- Preliminary Implementation Details
	- Separate dedicated page and table view. Similar to feature flags.

## Action Items

- [ ] @alexkim205 Basic wireframes by Wednesday standup (11/24)
- [ ] @alexkim205 Explore engineering constraints by Wednesday standup (11/24)

**Comment by marcushyett-ph** [comment]
@alexkim205  this is sounding great - can't wait to see some of this get shipped

One question - regarding integrations: What are your thoughts on embedding e.g. can someone embed a PostHog chart into a news article or a blog post etc? I've seen this as a popular format of sharing - e.g. twitter, Instagram etc - would love to hear your thoughts?

**Comment by pauldambra** [comment]
> can someone embed a PostHog chart into a news article or a blog post etc

What I've done elsewhere is generate an image on a timer (or on build for static sites) by rendering the web page and then capturing part of it as an image. You can then set that image as the image for the page in its OpenGraph data.

E.g. https://github.com/pauldambra/blog_source/blob/master/_plugins/create_week_note_image.rb

Which creates the image for e.g. shown in this toot https://twitter.com/pauldambra/status/1461996387827081218

(OpenGraph for that here: https://github.com/pauldambra/pauldambra.github.io/blob/5baa2f33a32a2177542ebf42486fb3bd701dcd25/weeknotes/2021/46.html#L48)

-----

That would only work for links that are publicly available like shared dashboards

I _think_ you'd need a slack app or similar so that people can share private dashboards and see a rendered image

**Comment by pauldambra** [comment]
TBH I'd bring at least some link usage metrics into scope. We'd want to see if they're being created and if they're being used to know whether to keep investing..?

**Comment by alexkim205** [comment]
> What are your thoughts on embedding e.g. can someone embed a PostHog chart into a news article or a blog post etc?

Yup, I think that's the right way to go about it. Insights and dashboards are the most insight-packed visuals in our app, so I think a snapshot of the insight would be a great start. 

re: Snapshots

That's pretty sweet! I wonder if for sharing private dashboards, we can have a placeholder image with just the insight name/description and a generic insight depending on what the insight type is. 

re: Link usage metrics

Sorry I should've expanded on that point. By link usage metrics I meant exposing a link's usage metrics in a modal somewhere so that users can see how many people clicked on it and viewed the insight. I think as per discussion [here](https://github.com/PostHog/posthog/issues/7244) it might be a stretch goal that's more tied to a centralized place to manage links. 

> TBH I'd bring at least some link usage metrics into scope. 

But definitely agree with this point. I think it's valuable to capture how many links are being shared and created (which I think we already do with `?ref=`). Might have to create a separate `?shareRef` parameter to track this feature specifically

**Comment by pauldambra** [comment]
> I wonder if for sharing private dashboards, we can have a placeholder image with just the insight name/description and a generic insight depending on what the insight type is.

If we ended up with a Teams app, Slack app, whatevs solution for private links. You could advertise that/those solutions in the generic image "You can't see that insight, but visit posthog.com/something-or-other to find out how you can"

**Comment by mariusandra** [comment]
re: Snapshots

- Let's remember that taking screenshots with a browser requires bundling a browser with the installation. This can get heavy, but may be unavoidable unless we use some "html to png" SaaS or microservice. 
- Alternative 1: we can re-implement all chart types in Python via [matplotlib](https://matplotlib.org/stable/gallery/index.html) that wi
- Alternative 2: we can complicate things further and re-implement all or most chart types with something like [react-svg-chart](https://www.npmjs.com/package/react-svg-chart), use `React.renderToString` via a nodejs server somewhere (could be inlined into django via [mini racer](https://pypi.org/project/py-mini-racer/)) to render the SVGs.
